packages/contentful-app-extensions/contributing-cohort-memberships/src/test/locations/Field.test.tsx

```diff
@@ -0,0 +1,266 @@
+describe('Field component', () => {
+  let sdk: jest.Mocked<FieldExtensionSDK>;
+
+  beforeEach(() => {
+    sdk = mockBaseSdk() as unknown as jest.Mocked<FieldExtensionSDK>;
+    (useSDK as jest.Mock).mockReturnValue(sdk);
+    (MultipleEntryReferenceEditor as jest.Mock).mockImplementation(() => (
+      <p>MultipleEntryReferenceEditor</p>
+    ));
+  });
+
+  it('enables automatic resizing', async () => {
+    render(<Field />);
+    expect(useAutoResizer).toHaveBeenCalled();
+  });
+
+  it('passes a custom card renderer to <MultipleEntryReferenceEditor />', async () => {
+    render(<Field />);
+
+    expect(MultipleEntryReferenceEditor).toHaveBeenCalled();
+    expect(
+      (MultipleEntryReferenceEditor as jest.Mock).mock.lastCall[0]
+        .renderCustomCard,
+    ).toEqual(CustomCard);
+  });
+
+  describe('CustomCard component', () => {
+    beforeEach(() => {
+      (useEntity as jest.Mock).mockImplementation((type, id) => {
+        if (id === 'contributing-cohort-1') {
+          return {
+            data: {
+              fields: {
+                name: {
+                  'en-US': 'My Contributing Cohort',
+                },
+              },
+            },
+          };
+        }
+      });
+    });
+
+    it('loads contributing cohort name from related entity and renders contributing cohort name and role', async () => {
+      const props = {
+        entity: {
+          fields: {
+            role: {
+              'en-US': 'Lead',
+            },
+            contributingCohort: {
+              'en-US': {
+                sys: {
+                  id: 'contributing-cohort-1',
+                },
+              },
+            },
+          },
+          sys: {
+            type: 'Entry',
+            publishedVersion: 1,
+            version: 1,
+          },
+        },
+        onEdit: jest.fn(),
+        onRemove: jest.fn(),
+      } as unknown as CustomEntityCardProps;
+
+      render(<CustomCard {...props} />);
+
+      await waitFor(() => {
+        expect(useEntity).toHaveBeenCalledWith(
+          'Entry',
+          'contributing-cohort-1',
+        );
+        expect(screen.getByText('My Contributing Cohort')).toBeInTheDocument();
+        expect(screen.getByText('Lead')).toBeInTheDocument();
+      });
+    });
+
+    it('displays an error message if no contributing cohort is selected', async () => {
+      const props = {
+        entity: {
+          fields: {
+            role: {
+              'en-US': 'Project Manager',
+            },
+            contributingCohort: null,
+          },
+          sys: {
+            type: 'Entry',
+            publishedVersion: 1,
+            version: 1,
+          },
+        },
+        onEdit: jest.fn(),
+        onRemove: jest.fn(),
+      } as unknown as CustomEntityCardProps;
+
+      render(<CustomCard {...props} />);
+
+      await waitFor(() => {
+        expect(
+          screen.getByText('No contributing cohort selected'),
+        ).toBeInTheDocument();
```

Comment:
use findByText in preference to waitFor and getByText. (Is it possible to use findByRole here?)